File: 백준/Silver/25192. 인사성 밝은 곰곰이/인사성 밝은 곰곰이.py
import sys

N = int(input())
answer = 0
name_set = set()

for _ in range(N):

    chat = sys.stdin.readline()
    # readline으로 읽으면 엔터까지 읽어서 비교할때 이렇게 해줘야한다.
    if chat == 'ENTER\n':
        name_set.clear() #clear함수 사용

    # 리스트에 값이 있는지 확인 : in 사용
    # 을 하면 시간복잡도가 O(N)으로 매우 느리게 동작한다.
    # 그래서 시간 초과를 막기 위해선 list말고 set을 사용해야한다.
    elif chat not in name_set:
        answer += 1
        name_set.add(chat)
print(answer)

File: 백준/Silver/25192. 인사성 밝은 곰곰이/인사성 밝은 곰곰이.cc
#include <bits/stdc++.h>
using namespace std;

int main(){
    ios_base::sync_with_stdio(false);
    cin.tie(0);

    set<string> gomgom;
    int N, cnt = 0;
    string chat;
    cin >> N;
    
    while(N--){
        cin >> chat;
        if(chat=="ENTER"){
            cnt += gomgom.size();
            gomgom.clear();
        }
        else{
            gomgom.insert(chat);
        }
    }
    cnt += gomgom.size();
    cout << cnt;
    
    return 0;
}